mlir/lib/Conversion/FuncToEmitC/FuncToEmitC.cpp

```diff
@@ -0,0 +1,119 @@
+//===- FuncToEmitC.cpp - Func to EmitC Patterns -----------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements patterns to convert the Func dialect to the EmitC
```

Comment:
This comment suggests that all of the func dialect operations are supported. This is not the case, as support for the `constant` and `call_indirect` operations is missing.

Reply:
I would read this as *This file implements patterns to convert **all operations of** the Func dialect to the EmitC* but I am open to suggestions.

Reply:
I don't have strong feelings about this, so I'm fine with keeping it as is.